src/dune_memory/dune_memory.ml

```diff
@@ -140,6 +145,16 @@ module Memory = struct
     let finally () = Stdune.Lock_file.unlock lock in
     Exn.protect ~f ~finally
 
+  let make_path memory path =
+    let path = Path.to_string path in
+    if Filename.is_relative path then
+      match memory.build_root with
+      | Some p ->
+        Result.ok (Path.of_string (Filename.concat (Path.to_string p) path))
+      | None -> Result.Error "relative path while no build root was set"
```

Comment:
Might be worth including the value of `path` in this error message to make debugging easier.